CONTRIBUTING.md

```diff
@@ -0,0 +1,138 @@
+# Code review checklist (TODO: should be copy-pasted in PR template)
+
+- [ ] code should be formatted using `rustfmt` (:warning: this also applied to code in macro like `tokio::select!` and `lazy_static!`
```

Comment:
I would rather say something like : the code in macros is not formatted, do your best to reduce the code size in macro arguments.

Reply:
```suggestion
- [ ] code should be formatted using `rustfmt` (:warning: this also applied to code in macro like `tokio::select!` and `lazy_static!`
- [ ] reduce the code size in macro arguments and in matching patterns
```

Reply:
@adrien-zinger I've taken the liberty to edit you suggestion as it contained only typos.